samza-core/src/main/java/org/apache/samza/container/grouper/task/BalancingTaskNameGrouper.java

```diff
@@ -54,5 +55,11 @@
    * @param localityManager provides a persisted task to container map to use as a baseline
    * @return                the grouped tasks in the form of ContainerModels
    */
-  Set<ContainerModel> balance(Set<TaskModel> tasks, LocalityManager localityManager);
+  default Set<ContainerModel> balance(Set<TaskModel> tasks, LocalityManager localityManager) {
+    return Collections.<ContainerModel>emptySet();
+  }
+  default Set<ContainerModel> balance(Set<Integer> containerIds, Set<TaskModel> tasks, LocalityManager localityManager) {
```

Comment:
I agree we should review if we need both of them. But for now I am just going to remove the unused balance() api. We will added it or merge with group() when we add Locality feature to standalone.